build-config/scripts/onie-mk-itb.sh

```diff
@@ -139,6 +141,7 @@ DTB="$(realpath $DTB)"
 			arch = "$ARCH";
 			os = "linux";
 			compression = "none";
+			load = <$FDT_LOAD>;
```

Comment:
I use $10 for FDT_LOAD, because $8 is used by SYSROOT.
diff --git a/build-config/arch/armv7a.make b/build-config/arch/armv7a.make
index 8c1bc86..4196d33 100644
--- a/build-config/arch/armv7a.make
+++ b/build-config/arch/armv7a.make
@@ -17,6 +17,7 @@ CROSSBIN    ?= $(XTOOLS_INSTALL_DIR)/$(TARGET)/bin
 KERNEL_ARCH            = arm
 KERNEL_LOAD_ADDRESS    ?= 0x60008000
 KERNEL_ENTRY_POINT     ?= 0x60008000
+FDT_LOAD_ADDRESS        ?= no
 KERNEL_DTB             ?= $(MACHINE).dtb
 KERNEL_DTB_PATH                ?= $(KERNEL_DTB)
 KERNEL_IMAGE_FILE      = $(LINUX_BOOTDIR)/compressed/piggy.gzip
diff --git a/build-config/arch/armv8a.make b/build-config/arch/armv8a.make
index a519ac0..e0227f7 100644
--- a/build-config/arch/armv8a.make
+++ b/build-config/arch/armv8a.make
@@ -17,6 +17,7 @@ CROSSBIN    ?= $(XTOOLS_INSTALL_DIR)/$(TARGET)/bin
 KERNEL_ARCH            = arm64
 KERNEL_LOAD_ADDRESS     ?= 0x80080000
 KERNEL_ENTRY_POINT      ?= 0x80080000
+FDT_LOAD_ADDRESS       ?= no
 KERNEL_DTB             ?= $(MACHINE).dtb
 KERNEL_DTB_PATH                ?= $(KERNEL_DTB)
 KERNEL_IMAGE_FILE      = $(LINUX_BOOTDIR)/Image.gz
diff --git a/build-config/arch/powerpc-softfloat.make b/build-config/arch/powerpc-softfloat.make
index 8d13f4c..13f2ed8 100644
--- a/build-config/arch/powerpc-softfloat.make
+++ b/build-config/arch/powerpc-softfloat.make
@@ -17,6 +17,7 @@ CROSSBIN    ?= $(XTOOLS_INSTALL_DIR)/$(TARGET)/bin
 KERNEL_ARCH            = powerpc
 KERNEL_LOAD_ADDRESS    ?= 0x00000000
 KERNEL_ENTRY_POINT     ?= 0x00000000
+FDT_LOAD_ADDRESS        ?= no
 KERNEL_DTB             ?= $(MACHINE).dtb
 KERNEL_DTB_PATH                ?= $(KERNEL_DTB)
 KERNEL_IMAGE_FILE      = $(LINUXDIR)/vmlinux.bin.gz
diff --git a/build-config/make/images.make b/build-config/make/images.make
index 958e156..17c8d2b 100644
--- a/build-config/make/images.make
+++ b/build-config/make/images.make
@@ -327,7 +327,8 @@ $(IMAGEDIR)/%.itb : $(KERNEL_INSTALL_STAMP) $(SYSROOT_CPIO_XZ) $(SCRIPTDIR)/onie
        $(Q) cd $(IMAGEDIR) && \
                V=$(V) $(SCRIPTDIR)/onie-mk-itb.sh $(MACHINE) $(MACHINE_PREFIX) $(UBOOT_ITB_ARCH) \
                $(KERNEL_LOAD_ADDRESS) $(KERNEL_ENTRY_POINT) \
-               $(KERNEL_VMLINUZ) $(IMAGEDIR)/$(MACHINE_PREFIX).dtb $(SYSROOT_CPIO_XZ) $@
+               $(KERNEL_VMLINUZ) $(IMAGEDIR)/$(MACHINE_PREFIX).dtb $(SYSROOT_CPIO_XZ) $@ \
+               $(FDT_LOAD_ADDRESS)

 $(UPDATER_ITB) : $(ITB_IMAGE)
        ln -sf $< $@
diff --git a/build-config/scripts/onie-mk-itb.sh b/build-config/scripts/onie-mk-itb.sh
index 8fa1c73..8d52508 100755
[zhaoq@mega:~/work/fork/onie/build-config]$git diff > patch
[zhaoq@mega:~/work/fork/onie/build-config]$cat patch
diff --git a/build-config/arch/armv7a.make b/build-config/arch/armv7a.make
index 8c1bc86..4196d33 100644
--- a/build-config/arch/armv7a.make
+++ b/build-config/arch/armv7a.make
@@ -17,6 +17,7 @@ CROSSBIN    ?= $(XTOOLS_INSTALL_DIR)/$(TARGET)/bin
 KERNEL_ARCH            = arm
 KERNEL_LOAD_ADDRESS    ?= 0x60008000
 KERNEL_ENTRY_POINT     ?= 0x60008000
+FDT_LOAD_ADDRESS        ?= no
 KERNEL_DTB             ?= $(MACHINE).dtb
 KERNEL_DTB_PATH                ?= $(KERNEL_DTB)
 KERNEL_IMAGE_FILE      = $(LINUX_BOOTDIR)/compressed/piggy.gzip
diff --git a/build-config/arch/armv8a.make b/build-config/arch/armv8a.make
index a519ac0..e0227f7 100644
--- a/build-config/arch/armv8a.make
+++ b/build-config/arch/armv8a.make
@@ -17,6 +17,7 @@ CROSSBIN    ?= $(XTOOLS_INSTALL_DIR)/$(TARGET)/bin
 KERNEL_ARCH            = arm64
 KERNEL_LOAD_ADDRESS     ?= 0x80080000
 KERNEL_ENTRY_POINT      ?= 0x80080000
+FDT_LOAD_ADDRESS       ?= no
 KERNEL_DTB             ?= $(MACHINE).dtb
 KERNEL_DTB_PATH                ?= $(KERNEL_DTB)
 KERNEL_IMAGE_FILE      = $(LINUX_BOOTDIR)/Image.gz
diff --git a/build-config/arch/powerpc-softfloat.make b/build-config/arch/powerpc-softfloat.make
index 8d13f4c..13f2ed8 100644
--- a/build-config/arch/powerpc-softfloat.make
+++ b/build-config/arch/powerpc-softfloat.make
@@ -17,6 +17,7 @@ CROSSBIN    ?= $(XTOOLS_INSTALL_DIR)/$(TARGET)/bin
 KERNEL_ARCH            = powerpc
 KERNEL_LOAD_ADDRESS    ?= 0x00000000
 KERNEL_ENTRY_POINT     ?= 0x00000000
+FDT_LOAD_ADDRESS        ?= no
 KERNEL_DTB             ?= $(MACHINE).dtb
 KERNEL_DTB_PATH                ?= $(KERNEL_DTB)
 KERNEL_IMAGE_FILE      = $(LINUXDIR)/vmlinux.bin.gz
diff --git a/build-config/make/images.make b/build-config/make/images.make
index 958e156..17c8d2b 100644
--- a/build-config/make/images.make
+++ b/build-config/make/images.make
@@ -327,7 +327,8 @@ $(IMAGEDIR)/%.itb : $(KERNEL_INSTALL_STAMP) $(SYSROOT_CPIO_XZ) $(SCRIPTDIR)/onie
        $(Q) cd $(IMAGEDIR) && \
                V=$(V) $(SCRIPTDIR)/onie-mk-itb.sh $(MACHINE) $(MACHINE_PREFIX) $(UBOOT_ITB_ARCH) \
                $(KERNEL_LOAD_ADDRESS) $(KERNEL_ENTRY_POINT) \
-               $(KERNEL_VMLINUZ) $(IMAGEDIR)/$(MACHINE_PREFIX).dtb $(SYSROOT_CPIO_XZ) $@
+               $(KERNEL_VMLINUZ) $(IMAGEDIR)/$(MACHINE_PREFIX).dtb $(SYSROOT_CPIO_XZ) $@ \
+               $(FDT_LOAD_ADDRESS)

 $(UPDATER_ITB) : $(ITB_IMAGE)
        ln -sf $< $@
diff --git a/build-config/scripts/onie-mk-itb.sh b/build-config/scripts/onie-mk-itb.sh
index 8fa1c73..8d52508 100755
--- a/build-config/scripts/onie-mk-itb.sh
+++ b/build-config/scripts/onie-mk-itb.sh
@@ -23,7 +23,8 @@ MACHINE_PREFIX="$2"

 ARCH="$3"
 if [ "$ARCH" != "ppc" ] &&
-   [ "$ARCH" != "arm" ] ; then
+   [ "$ARCH" != "arm" ] &&
+   [ "$ARCH" != "arm64" ] ; then
     echo "Error: Unsupported architecture: $ARCH"
     exit 1
 fi
@@ -40,6 +41,11 @@ KERNEL_ENTRY="$5"
     exit 1
 }

+FDT_LOAD="$10"
+[ -n "$FDT_LOAD" ] || {
+    echo "Error: FDT_LOAD was not specified"
+    exit 1
+}
 KERNEL_COMPRESSION="gzip"
 INITRD_LOAD="0x00000000"
 FDT="fdt = \"dtb\";"
@@ -85,6 +91,8 @@ KERNEL="$(realpath $KERNEL)"
 SYSROOT="$(realpath $SYSROOT)"
 DTB="$(realpath $DTB)"

+cat $its_file
+
 # Create a .its file for this machine type on the fly
 (cat <<EOF
 /*
@@ -139,6 +147,17 @@ DTB="$(realpath $DTB)"
                        arch = "$ARCH";
                        os = "linux";
                        compression = "none";
+EOF
+) > $its_file
+
+if [ "$FDT_LOAD" != "no" ] ; then
+(cat <<EOF
+                       load = <$FDT_LOAD>;
+EOF
+) >> $its_file
+fi
+
+(cat <<EOF
                        hash@1 {
                                algo = "crc32";
                        };
@@ -158,7 +177,9 @@ DTB="$(realpath $DTB)"
 };

 EOF
-) > $its_file
+) >> $its_file
+
+cat $its_file

 if [ "$V" != "0" ] ; then
     echo "=========================================="
diff --git a/machine/nxp/nxp_ls1043ardb/machine.make b/machine/nxp/nxp_ls1043ardb/machine.make
index 7a1be31..2544ac5 100644
--- a/machine/nxp/nxp_ls1043ardb/machine.make
+++ b/machine/nxp/nxp_ls1043ardb/machine.make
@@ -20,6 +20,8 @@ UBOOT_MACHINE = ls1043ardb_ONIE_$(MACHINE_REV)
 KERNEL_DTB = freescale/fsl-ls1043a-rdb.dtb
 KERNEL_DTB_PATH = dts/$(KERNEL_DTB)

+FDT_LOAD_ADDRESS = 0x90000000
+
 # Vendor ID -- IANA Private Enterprise Number:
 # http://www.iana.org/assignments/enterprise-numbers
 VENDOR_ID = 33118